server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckBaseUtility.java

```diff
@@ -230,4 +230,46 @@ protected void checkDuplicates(
     updatedDuplicateOptions.putAll(customDuplicateOptions);
     checkDuplicateEntries(ctx, foundEntries, updatedDuplicateOptions);
   }
+
+  /**
+   * Flags errors for rule lists passed as parameters if there are multiple instances of mutually
+   * exclusive options.
+   *
+   * @param options Options checked to insert into error message
+   * @param rules Generic lists of rule lists to check. Will be a collection of ParserRuleContext
+   *     and/or TerminalNode objects.
+   * @param <E> List of Generics to allow cross-rule context collection.
+   * @return True if an instance is not found
+   */
+  protected <E extends ParseTree> boolean checkHasMutuallyExclusiveOptions(
+      String options, List<E>... rules) {
+    String token = null;
+    for (List<E> option : rules) {
+      if (option.isEmpty()) continue;
+      if (token == null) token = option.remove(0).getText().split("\\(")[0].toUpperCase();
+      for (E rule : option) {
+        // Flag only options that do NOT match the first option encountered.
+        // If the option is the same then this is already flagged by checkDuplicates
+        if (!token.equals(rule.getText().split("\\(")[0].toUpperCase())) {
+          throwException(
+              ErrorSeverity.ERROR,
+              getLocality(rule),
+              "Exactly one option required, options are mutually exclusive: ",
+              options);
+        }
+      }
+    }
+    return token == null;
```

Comment:
This is a bit confusing. The other check functions return true when "check passed".
I suggest returing e.g. number of alternatives detected, so that it is clear the return value is different.